semgrep-core/src/parsing/Parse_rule.ml

```diff
@@ -1402,11 +1402,11 @@ let parse_file ?error_recovery file =
               if n <> 0 then failwith (spf "error executing %s" cmd);
               let ast = Parse_json.parse_program tmpfile in
               Json_to_generic.program ~unescape_strings:true ast)
-    | FT.Config FT.Yaml -> parse_yaml_rule_file file
+    | FT.Config FT.Yaml -> parse_yaml_rule_file ~is_target:false file
```

Comment:
this should be true

Reply:
Why did you continue to use false?
We're parsing a rule, so the rule will not contain metavariables; only the patterns inside the rule will, but that's using a different function (Parse_pattern).